homeassistant/components/aosmith/config_flow.py

```diff
@@ -0,0 +1,99 @@
+"""Config flow for A. O. Smith integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from py_aosmith import AOSmithAPIClient, AOSmithInvalidCredentialsException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for A. O. Smith."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Start the config flow."""
+        self._reauth_entry = None
+        self._email = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            unique_id = user_input[CONF_EMAIL].lower()
```

Comment:
Also, can you change your email?

Reply:
I checked the mobile app and it seems it's not possible to change your email.